src/Compilers/Core/Portable/CommandLine/EditorConfig.SectionNameMatching.cs

```diff
@@ -179,7 +316,38 @@ public TokenKind Lex()
             /// <summary>
             /// Call after getting <see cref="TokenKind.SimpleCharacter" /> from <see cref="Lex()" />
             /// </summary>
-            public char EatCurrentCharacter() => _sectionName[_position++];
+            public char EatCurrentCharacter() => _sectionName[Position++];
+
+            public char this[int position] => _sectionName[position];
+
+            /// <summary>
+            /// Returns the string representation of a decimal integer, or null if
+            /// the current lexeme is not an integer.
+            /// </summary>
+            public string TryLexNumber()
```

Comment:
Given we are using `ref struct` here already did you consider defining a `ref struct` to represent a text range here?

Reply:
Seemed like a tuple was sufficient here, since it's just a pair.

Reply:
Don't quite understand your response here as there is no tuple. 

---
In reply to: [199956869](https://github.com/dotnet/roslyn/pull/28083#discussion_r199956869) [](ancestors = 199956869)

Reply:
Oh, never mind, wrong method. What would be in the ref-struct? Just a string? I didn't think that was necessary.

Reply:
Basically equivalent of `Span<char>`. It's not strictly necessary here. Given you were using ref struct I watned to know if it's an option you considered.

---
In reply to: [201843061](https://github.com/dotnet/roslyn/pull/28083#discussion_r201843061) [](ancestors = 201843061)